### PR TITLE
Show `link-out-of-scope` warning when link to a file outside initial build scope

### PR DESCRIPTION
**Pull request by yufeih**
Handles file outside initial build scope

#2699

### CODE REVIEW

**Comment by herohua** [comment]
@live1206

File: docs/specs/resolve.yml [hunk 1]
```diff
@@ -219,4 +219,61 @@ inputs:
...
+inputs:
+  docfx.yml: |
+    dependencies:
+      dep: https://github.com/docascode/docfx-test-dependencies
+  docs/a.md: 'Link to [dep](~/dep/logo.svg)'
+outputs:
+  docs/a.json: |
+    {"content":"<p>Link to <a href=\"%7E/dep/logo.svg\">dep</a></p>"}
+  build.log: |
+    ["warning","link-is-dependency","File 'logo.svg' referenced by link '~/dep/logo.svg' will not be build because it is from a dependency docset","docs/a.md"]
```
Comment (superyyrrzz) [inline docs/specs/resolve.yml]: >build [](start = 101, length = 5)

built

File: src/docfx/docset/Docset.cs [hunk 2]
```diff
@@ -28,27 +29,37 @@ internal class Docset
...
+        public Docset(Context context, string docsetPath, Config config, CommandLineOptions options)
         {
             DocsetPath = Path.GetFullPath(docsetPath);
             Config = config;
-            CombinedRedirections = CombineRedirections(config);
 
             // pass on the command line options to its children
             _options = options;
+            _context = context;
+            _buildScope = new Lazy<HashSet<Document>>(() => GlobFiles());
```
Comment (OsmondJiang) [inline src/docfx/docset/Docset.cs]: > GlobFiles() [](start = 59, length = 12)

pass the redirections as parameter to clear the interface

File: src/docfx/build/resolve/Resolve.cs [hunk 3]
```diff
@@ -104,7 +116,7 @@ private static (DocfxException error, Document file, string redirectTo, string f
             // resolve from redirection files
             pathToDocset = PathUtility.NormalizeFile(pathToDocset);
 
-            if (relativeTo.Docset.CombinedRedirections.TryGetValue(pathToDocset, out var redirectTo))
+            if (relativeTo.Docset.Redirections.TryGetRedirectionUrl(pathToDocset, out var redirectTo))
             {
                 // redirectTo always is absolute href
                 return (null, null, redirectTo, null, null);
```
Comment (OsmondJiang) [inline src/docfx/build/resolve/Resolve.cs]: please add TODO to return error if the redirectTO doesn't belong to build scope

**Pull request merged**